tests/lib/MVC/Symfony/EventListener/SiteAccessMatchListenerTest.php

```diff
@@ -155,6 +182,38 @@ public function testOnKernelRequestSerializedSAWithCompoundMatcher()
         $this->assertFalse($request->attributes->has('serialized_siteaccess'));
     }
 
+    public function testOnKernelRequestSerializedSAWithMatcherInMatcherRegistry(): void
+    {
+        $matcher = new CustomMatcher([]);
+        $matcherRegistryMock = $this->createMock(SiteAccessMatcherRegistryInterface::class);
+        $matcherRegistryMock->method('hasMatcher')->willReturn(true);
+        $matcherRegistryMock->method('getMatcher')->willReturn($matcher);
+        $matcher2 = new CustomMatcher([]);
+        $matcher2->setMapKey('key_foobar');
+
+        $matcherRegistryMock
+            ->expects($this->once())
+            ->method('getMatcher')
+            ->with('Ibexa\Tests\Core\MVC\Symfony\EventListener\CustomMatcher');
+        $this->listener = new SiteAccessMatchListener(
+            $this->saRouter,
+            $this->eventDispatcher,
+            $matcherRegistryMock
+        );
+
+        $siteAccess = $this->createSiteAccess(
+            $matcher2,
+            null,
+            [new SiteAccessGroup('test_group')]
+        );
+
+        $request = $this->createRequest($siteAccess);
+        /** @var CustomMatcher $siteAccessMatcher */
+        $siteAccessMatcher = $siteAccess->matcher;
+        $this->assertEquals('key_foobar', $siteAccessMatcher->getMapKey());
+        $this->assertFalse($request->attributes->has('serialized_siteaccess'));
```

Comment:
CS: static methods need to be called statically:
```suggestion
        self::assertEquals('key_foobar', $siteAccessMatcher->getMapKey());
        self::assertFalse($request->attributes->has('serialized_siteaccess'));
```